internal/pkg/api/vsphere.go

```diff
@@ -176,15 +176,15 @@ func WithDatacenter(value string) VSphereFiller {
 	}
 }
 
-func WithStringFromEnvVar(envVar string, opt func(string) VSphereFiller) VSphereFiller {
+func WithVSphereStringFromEnvVar(envVar string, opt func(string) VSphereFiller) VSphereFiller {
```

Comment:
There is no need for this to be VSphere specific right? We have have it generic with an interface I believe, so that we don't unnecessarily maintain these functions in two places.

Reply:
Is it already implemented? From what I can tell, we're returning a VSphereFiller type. What's your suggestion here?

Reply:
Now that I look at the code more, we may need to make an interface that can handle `VSphereConfig` and `CloudStackConfig` and make `VSphereFiller` more generic like `ProviderFiller`, but that would probably be better for a follow up. I think this is fine for now.

Reply:
Created https://github.com/aws/eks-anywhere/issues/1758, feel free to add additional details